### PR TITLE
Add support for TT Llama3 text models (1B,3B,8B,70B-new)

### PR DESCRIPTION
**Pull request by skhorasganiTT**
- Add support for remaining llama3 TT text models (1B,3B,8B, and new 70B) in TT backend and inference/server examples
- Add supported model check to inference scripts
- Add TG device type to TTWorker
- Remove custom Llama70b model execution code in TTModelRunner and replace with commonized forward functions
- Note: this PR relies on https://github.com/tenstorrent/tt-metal/pull/16292

cc @cglagovichTT

### CODE REVIEW

**Review by uaydonat — APPROVED** [review]

File: tt_metal/README.md [hunk 1]
```diff
@@ -46,40 +47,46 @@ To run Meta-Llama-3.1/3.2, it is required to have access to the model on Hugging
...
 ```
 
-To measure performance for a single batch (with the default prompt length of 128 tokens):
+To measure performance (Llama70B) for a single batch (with the default prompt length of 128 tokens):
 ```python
-WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml python examples/offline_inference_tt.py --measure_perf
+MESH_DEVICE=T3K_LINE WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml python examples/offline_inference_tt.py --measure_perf
 ```
 
+**Note**: By default, the inference example will run with Llama-3.1-70B. To run with Llama-3.1-8B, Llama-3.2-1B, or Llama-3.2-3B, ensure that the apprioriate environement variables are set as per the [demo instructions](https://github.com/tenstorrent/tt-metal/tree/main/models/demos/llama3), then set `MESH_DEVICE=N150` and one of the following:
```
Comment (uaydonat) [inline tt_metal/README.md:66]: other models only work with N150, not other systems?

File: tt_metal/README.md [hunk 2]
```diff
@@ -35,7 +35,8 @@ Git-checkout the following branches in each repo separately:
 To run Meta-Llama-3.1/3.2, it is required to have access to the model on Hugging Face. To gain access:
 1. Request access on Hugging Face:
     - Llama-3.1: [https://huggingface.co/meta-llama/Meta-Llama-3.1-70B](https://huggingface.co/meta-llama/Meta-Llama-3.1-70B)
-    - Llama-3.2: [https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct](https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct)
+    - Llama-3.2: [https://huggingface.co/meta-llama/Llama-3.2-1B](https://huggingface.co/meta-llama/Llama-3.2-1B)
```
Comment (uaydonat) [inline tt_metal/README.md:38]: where are 1B. 3B and 8B?

File: tt_metal/README.md (hunk 1 above)
Comment (uaydonat) [inline tt_metal/README.md:66]: typo "environement"

File: tt_metal/README.md [hunk 3]
```diff
@@ -46,40 +47,46 @@ To run Meta-Llama-3.1/3.2, it is required to have access to the model on Hugging
...
 
 ## Running the offline inference example
 
-### Llama-3.1-70B
+### Llama-3.1/3.2 Text Models (1B, 3B, 8B, 70B)
 
-To generate tokens for sample prompts:
+To generate tokens (Llama70B) for sample prompts:
 ```python
-WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml python examples/offline_inference_tt.py
```
Comment (uaydonat) [inline tt_metal/README.md:57]: what's the batch size for this command? Maybe add to notes.

File: tt_metal/README.md [hunk 4]
```diff
@@ -46,40 +47,46 @@ To run Meta-Llama-3.1/3.2, it is required to have access to the model on Hugging
...
+MESH_DEVICE=T3K_LINE WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml python examples/offline_inference_tt.py
 ```
 
-To measure performance for a single batch (with the default prompt length of 128 tokens):
+To measure performance (Llama70B) for a single batch (with the default prompt length of 128 tokens):
```
Comment (uaydonat) [inline tt_metal/README.md:61]: why single batch? I cannot measure with batch=32? If this is configurable, let's add the option to notes.

File: tt_metal/README.md (hunk 2 above)
Comment (skhorasganiTT) [reply tt_metal/README.md:38]: Gaining access to either of the 3.1 models (70B or 8B) will provide access for both, and the same applies for 3.2 (using the link to apply for 1B will also provide 3B)

File: tt_metal/README.md (hunk 4 above)
Comment (skhorasganiTT) [reply tt_metal/README.md:61]: Single batch refers to one batch with batch_size=32. I've updated the readme to make it more clear

File: tt_metal/README.md (hunk 3 above)
Comment (skhorasganiTT) [reply tt_metal/README.md:57]: The default batch size is 32, I've added the note to the readme

File: tt_metal/README.md (hunk 1 above)
Comment (skhorasganiTT) [reply tt_metal/README.md:66]: fixed

Comment (skhorasganiTT) [reply tt_metal/README.md:66]: They work with other systems as well, I have updated the readme

**Pull request merged**